modules/apps/frontend-data-set/frontend-data-set-admin-web/src/main/resources/META-INF/resources/item/selector/FDSAdminItemSelector.tsx

```diff
@@ -25,8 +25,39 @@ const views = [
 		name: 'list',
 		schema: {
 			description: 'description',
+			sticker: 'sticker',
 			symbol: 'symbol',
 			title: 'label',
+			tooltip: 'tooltip',
+		},
+		setItemComponentProps: ({item, props}: {item: any; props: any}) => {
+			if (
+				!item.dataSetToDataSetCardsSections.length &&
+				!item.dataSetToDataSetTableSections.length &&
+				!item.dataSetToDataSetListSections.length
+			) {
+				return {
+					...props,
+					item: {
+						...item,
+						sticker: {displayType: 'warning'},
+						symbol: 'exclamation-circle',
+						tooltip: Liferay.Language.get(
+							'no-visualization-modes-have-been-defined'
+						),
```

Comment:
:rocket: With this change, now the concept of "valid" is meaningless for FDS, which limits itself to render what's told to. Parent component here is in charge of defining the intent of the item depending on custom conditions.

This is the right pattern!